debugAdapterProtocol.json

```diff
@@ -808,6 +808,66 @@
 				"required": [ "body" ]
 			}]
 		},
+		"StartDebuggingRequest": {
+			"allOf": [
+				{
+					"$ref": "#/definitions/Request"
+				},
+				{
+					"type": "object",
+					"description": "This request is sent from the debug adapter to the client to start a new debug session of the same type as the caller.\nThis event should only be sent if the corresponding client capability `supportsStartDebuggingRequest` is true.\nA client implementation of `startDebugging` should start a new debug session (of the same type as the caller) in the same way that the caller's session was started. If the client supports hierarchical debug sessions, the newly created session can be treated as a child of the caller session.",
+					"properties": {
+						"command": {
+							"type": "string",
+							"enum": [
+								"startDebugging"
+							]
+						},
+						"arguments": {
+							"$ref": "#/definitions/StartDebuggingRequestArguments"
+						}
+					},
+					"required": [
+						"command",
+						"arguments"
+					]
+				}
+			]
+		},
+		"StartDebuggingRequestArguments": {
+			"type": "object",
+			"description": "Arguments for `startDebugging` request.",
+			"properties": {
+				"configuration": {
+					"type": "object",
+					"additionalProperties": true,
+					"description": "Arguments passed to the new debug session. The arguments must only contain properties understood by the `launch` or `attach` requests of the debug adapter and they must not contain any client-specific properties (e.g. `type`) or client-specific features (e.g. substitutable 'variables')."
+				},
+				"request": {
+					"type": "string",
+					"enum": [
+						"launch",
+						"attach"
+					],
+					"description": "Indicates whether the new debug session should be started with a `launch` or `attach` request."
```

Comment:
Perhaps we should additionally note that the value here does not change _how_ the client launches the new session (it's launched in the same way as the calling session) but is only used for UI.

Reply:
Discussed offline- this does determine whether the session is started with an 'attach' or 'launch' request.